blockchain/blocks/src/header.rs

```diff
@@ -201,6 +202,22 @@ impl<'de> Deserialize<'de> for BlockHeader {
     }
 }
 
+impl Ord for BlockHeader {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.ticket().cmp(other.ticket()) {
+            // Only compare cid bytes when tickets are equal
+            Ordering::Equal => self.cid().to_bytes().cmp(&other.cid().to_bytes()),
+            order => order,
+        }
```

Comment:
```suggestion
        // Only compare cid bytes when tickets are equal
        self.ticket()
            .cmp(other.ticket())
            .then_with(|| self.cid().to_bytes().cmp(&other.cid().to_bytes()))
```
Good stuff, just wanted to bring to your attention the `Ordering::{then, then_with}` methods that serve exactly the purpose of composing orderings